meetings/2021-07-21.md

```diff
@@ -0,0 +1,176 @@
+
+## Present
+
+* Henry Story
+* Justin Bingham
+* Elf Pavlik
+* Matthieu B
+
+
+## Minutes
+
+### ACP UC3
+
+Henry: I don't think there is a way to build Solid without allowing global read to ACRs. There has to be some document that tells you how to authenticate. Now, it is true that there is a privacy problem with people putting all access control statements in one ACL (a problem that the current WAC `:default` behavior leads to) as there is no way then to set access control rules on the rules. But that problem can be made to disappear, as shown in [comment on issue 189](https://github.com/solid/authorization-panel/issues/189#issuecomment-797638239). If the client cannot know what the rules are, it cannot know how to access a resource. This would then be a major missing piece for solid to work at all.  If we don't see this as working, we would need to inventing a whole new access control system. 
+
+But we are getting there. This is why it is important to go through those Use Cases.
+I don't think we want to go through all Use Cases though, as that will take too much time. They were developed before we went through this process. 
+I don't see that the [latest comment on UC3](https://github.com/solid/authorization-panel/pull/216#discussion_r673945084) is a problem, it is up to Elf to put a proposition forward to deal with credentials requirement discovery.
+
+The default is that you have to read the access control. (Henry: I later add some nuance to this: this is not to say that the ACL has to by default have public access - it may be important to explicitly state so)
+What happens if you don't get access to the ACL? You don't know how to authenticate: therefore, you don't have access.  You would be left with guessing, which is not acceptable for privacy reasons on the client.
+
+Matthieu: But it is a security issue to have ACRs readable publicly by default.
+
+Henry: It depends how it is done. 
+It can be done right, and in my view, far from being an exceptional case, it is going to be the majority of cases that do this.
+
+Justin: I think we're conflating a real problem that needs to be solved: which is understanding what a given agent has access to.  An access control system should just be telling us whether you have access to something or not. But there are other out-of-band ways to give agents a list of what they have access to.
+There aren't patterns to tell an agent what they have access to.
+I personally don't believe that the best way to deliver that information is through the ACL resources. I'm happy to be proven wrong.
+I see it as a separate type of flow.
+In other areas which aren't Solid specific but if your control rules are setup (OAuth/Firewalls...).
+I'm wondering what is the minimum amount that people need to know.
+We have precedent with WAC-ALLOW in Solid.
+But we need to be able to follow your nose through resources would be pretty problematic. Because you don't want to leak stuff.
```

Comment:
I think the idea behind that statement is:

> Using a follow your nose mechanism, that is requiring the dereferencing of access control resources to understand access requirement would be pretty problematic. Because you don't want to leak information.

Reply:
Tweaked it a bit; updated suggestion.